ffi/derive/src/convert.rs

```diff
@@ -490,3 +529,30 @@ fn is_opaque_wrapper(input: &DeriveInput) -> bool {
     let opaque_attr = parse_quote! {#[opaque_wrapper]};
     input.attrs.iter().any(|a| *a == opaque_attr)
 }
+
+fn add_bounds_and_split_generics<'generics>(
+    generics: &'generics mut syn::Generics,
+    lifetimes: &'generics [syn::Lifetime],
+    trait_bounds: &'generics [syn::TraitBound],
+) -> (
+    syn::punctuated::Punctuated<syn::GenericParam, syn::Token![,]>,
+    syn::TypeGenerics<'generics>,
+    Option<&'generics syn::WhereClause>,
+) {
+    generics
+        .lifetimes_mut()
+        .for_each(|lt| lt.bounds.extend(lifetimes.iter().cloned()));
+    generics.type_params_mut().for_each(|param| {
+        param
+            .bounds
+            .extend(lifetimes.iter().cloned().map(syn::TypeParamBound::from));
+        param
+            .bounds
+            .extend(trait_bounds.iter().cloned().map(syn::TypeParamBound::from));
+    });
```

Comment:
so you need the `<'itm>` lifetime bound on every generic parameter?

Reply:
@mversic, I think so, `TryFromReprC<'itm>: Sized + 'itm` so if `Type: TryFromRepr<'itm>` then `Type: 'itm` which should imply that every generic parameter of `Type` should live for at least `'itm`.

Small [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0a3d2a2ee833a5d8992033054171de4d).

Reply:
yes, it's good as is, you can also set a bound on the whole type and the bound will work recursively on inner types. Let me save you the trouble [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d9feab412c1eb20f0842c9b8e184125d)

Reply:
wow, cool! i don't know that this will work :) this allow write function in really clear way